File: app/src/main/java/de/garten/training/depotflow/data/api/dto/WorkOrderDto.java
package de.garten.training.depotflow.data.api.dto;

import java.util.List;

public class WorkOrderDto {
    public String id;
    public String externalNumber;
    public String title;
    public String customerName;
    public String status;
    public int priority;
    public String dueAt;
    public String updatedAt;
    public String assignedUser;
    public List<StopDto> stops;
    public List<ChecklistItemDto> checklist;

    public WorkOrderDto() {
    }

    public WorkOrderDto(String id, String externalNumber, String title, String customerName, String status, int priority) {
        this.id = id;
        this.externalNumber = externalNumber;
        this.title = title;
        this.customerName = customerName;
        this.status = status;
        this.priority = priority;
    }
}

File: app/src/main/java/de/garten/training/depotflow/data/api/dto/WorkOrderDto.kt
package de.garten.training.depotflow.data.api.dto

class WorkOrderDto {
    @JvmField
    var id: String? = null

    @JvmField
    var externalNumber: String? = null

    @JvmField
    var title: String? = null

    @JvmField
    var customerName: String? = null

    @JvmField
    var status: String? = null

    @JvmField
    var priority: Int = 0

    @JvmField
    var dueAt: String? = null

    @JvmField
    var updatedAt: String? = null

    @JvmField
    var assignedUser: String? = null

    @JvmField
    var stops: List<StopDto>? = null

    @JvmField
    var checklist: List<ChecklistItemDto>? = null

    constructor()

    constructor(
        id: String?,
        externalNumber: String?,
        title: String?,
        customerName: String?,
        status: String?,
        priority: Int
    ) {
        this.id = id
        this.externalNumber = externalNumber
        this.title = title
        this.customerName = customerName
        this.status = status
        this.priority = priority
    }
}
